test/runtests.jl

```diff
@@ -111,5 +111,53 @@ using Test
             @test call1 != call3
             @test call2 != call4
         end
+
+        @testset "splatting" begin
+            f(a, b; kw1=1, kw2=2) = a + b + kw1 + kw2
+
+            one = 1
+            a = [1, 2]
+            kw = (kw1=1, kw2=2)
+
+            @testset "basic splatting" begin
+                @test f(a...) == @cached_call f(a...)
+                @test f(a...) == @cached_call f([1, 2]...)
+                @test f(a...) == @cached_call f([one, 2]...)
+                @test f(a...; kw...) == @cached_call f(a...; kw...)
+                @test f(a...; kw...) == @cached_call f(a...; (kw1=1, kw2=2)...)
+                @test f(a...; kw...) == @cached_call f(a...; (kw1=one, kw2=2)...)
+
+                @test @hash_call(f(a...)) isa UInt
```

Comment:
only test the type of `@hash_call`? Do you want to test the exact value relationship under various situations?

Reply:
`@cached_call` calls the `@hash_call` macro under the hood, which means that most of the functionality is tested via `@cached_call` tests. However, the expression passed to `@cached_call` is escaped (macro hygiene) before being passed to `@hash_call`. Since `@hash_call` is exported we also want to test it parses non-escaped expressions correctly and can be used directly. This is what these tests are here for.

(I should write this down as a comment)